src/material/snack-bar/snack-bar-container.ts

```diff
@@ -220,6 +231,54 @@ export abstract class _MatSnackBarContainerBase extends BasePortalOutlet impleme
         element.classList.add(panelClasses);
       }
     }
+
+    this._exposeToModals();
+  }
+
+  /**
+   * Some browsers won't expose the accessibility node of the live element if there is an
+   * `aria-modal` and the live element is outside of it. This method works around the issue by
+   * pointing the `aria-owns` of all modals to the live element.
+   */
+  private _exposeToModals() {
+    // TODO(crisbeto): consider de-duplicating this with the `LiveAnnouncer`.
+    // Note that the selector here is limited to CDK overlays at the moment in order to reduce the
+    // section of the DOM we need to look through. This should cover all the cases we support, but
+    // the selector can be expanded if it turns out to be too narrow.
+    const id = this._liveElementId;
+    const modals = this._document.querySelectorAll(
+      'body > .cdk-overlay-container [aria-modal="true"]',
+    );
+
+    for (let i = 0; i < modals.length; i++) {
```

Comment:
Nit: Could use `for (const modal of modals)`

Reply:
That doesn't work, because `querySelectorAll` returns a `NodeList`.

Reply:
A node list is iterable, so it should work: https://developer.mozilla.org/en-US/docs/Web/API/NodeList

But no need to follow-up. As said, a nit.